Index: ioxpc.cpp
===================================================================
--- ioxpc.cpp   (revision 787)
+++ ioxpc.cpp   (working copy)
@@ -159,6 +159,7 @@
       r = xpcu_request_28(xpcu, 0x11);
       if (r>=0) r = xpcu_output_enable(xpcu, 1);
       if (r>=0) r = xpcu_shift(xpcu, 0xA6, 2, 2, zero, 0, NULL);
+      if (r>=0) r = xpcu_shift(xpcu, 0xA6, 2, 2, zero, 0, NULL);
       if (r>=0) r = xpcu_request_28(xpcu, 0x12);
       
       if (r<0)

